quickfixj-core/src/main/java/quickfix/Session.java

```diff
@@ -881,7 +881,6 @@ public void reset() {
                     ((ApplicationExtended) application).onBeforeSessionReset(sessionID);
                 }
                 state.setResetStatePending(true);
-                logout("Session reset");
                 getLog().onEvent("Initiated logout request");
                 generateLogout(state.getLogoutReason());
```

Comment:
```suggestion
                generateLogout("Session reset");
                getLog().onEvent("Initiated logout request");
```
I've swapped two lines here since the Logout is now initiated after calling `generateLogout()`. Moreover I've directly added the logout reason to `generateLogout()` since that would have been set in `logout("Session reset")` before.

NB: Actually I am asking myself if the former code with call to `logout()` AND `generateLogout()` could have lead to a double Logout (or at least the possibility) since `generateLogout()` and `next()` could have created a Logout.